### PR TITLE
DEV tools: Script to automate creation of example code for an IC

### PR DESCRIPTION
**Pull request by ragv**
**1. Created a new folder `/dev/`**
**2. Added `IC_example_code_maker.py`\- a script to automate the process of making an example code for an IC**
- Takes IC No. as Input
- Scrapes the tests.py file for the sample input case
- uses `exec` to check if IC present in the code base.
- Extensively uses `format` to generate the code from a template code.
- The synthesized code is run to verify the integrity.
- produces an output ICNO.py in the dev folder . Has to be manually moved inside example.py
  MOTIVATION: Saves a lot of dev time.

### CODE REVIEW

**Comment by ghost** [comment]
@jayrambhia pl review ! :)

File: BinPy/dev/IC_example_code_maker.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+        print 'Dictinary of variables to replace in the template code',dict_of_vars
+
+        code =  "from BinPy import *\n"
+        code += "print 'Usage of IC {IC_NO}:\\n'\n".format(**dict_of_vars)
+        code += "ic = IC_{IC_NO}()\n".format(**dict_of_vars)
+        code += "print '\\nThe Pin configuration is:\\n'\n"
+        code += "p = {PIN_CONF}\n".format(**dict_of_vars)
+        code += "print p\n"
+        code += "print '\\nPin initinalization\\n'\n"
```
Comment (rajat974) [inline BinPy/dev/IC_example_code_maker.py:56]: please change it again :)

**Comment by jayrambhia** [comment]
@rvraghav93  give me some time more time to think about this.

File: BinPy/dev/IC_example_code_maker.py (hunk 1 above)
Comment (ragv) [inline BinPy/dev/IC_example_code_maker.py:56]: @rajat974 Do you mean the print statement that returns None ? I corrected it... see the next line !! :) If you find any other typos let me know ! :)

**Pull request closed without merging**